### PR TITLE
Signup: Remove main flow

### PR DESCRIPTION
**Pull request by arthur791004**
#### Changes proposed in this Pull Request

* Remove `main` flow as it seems not to get any traffic. Also for `about` step
* Besides, remove the following items which are only used by `about` step
  * `site-goals` related code from `calypso/state/signup/steps/site-goals`
  * `user-experience` related code from `calypso/state/signup/steps/user-experience`
  * `getThemeForSiteGoals` and `getDesignTypeForSiteGoals` utils

#### Testing instructions

* Go to `/start`
* Check existing flows work well

Related to https://github.com/Automattic/wp-calypso/issues/60281

### CODE REVIEW

**Comment by p-jackson** [comment]
There was a time when the wpcom desktop app was using the `main` flow ... P7jrAc-3-p2
Still, the traffic looks like it was very very low. I'll test out the desktop app and see what I see.

**Comment by arthur791004** [comment]
I think the default screen on desktop app is [log-in](https://github.com/Automattic/wp-calypso/blob/5f4eb761353006b94555609175101ac4f95d7ac1/desktop/app/lib/config/index.js#L11)

image

When you click “Create a new account”, the flow is `onboarding`
image

When you click “Sign Up” button at the top-right corner, the flow is also `onboarding`
image

So, I think desktop app doesn't use `main` flow now.

**Comment by p-jackson** [comment]
Tested and I didn't see it either. Looks like things have changed a bit since I last looked at how wp-desktop worked pbAok1-2hj-p2#comment-4640

File: client/signup/steps/plans/index.jsx [hunk 1]
```diff
@@ -94,10 +92,7 @@ export class PlansStep extends Component {
 			return this.props.customerType;
 		}
 
-		const siteGoals = this.props.siteGoals.split( ',' );
-		const customerType =
-			getSiteTypePropertyValue( 'slug', this.props.siteType, 'customerType' ) ||
-			( intersection( siteGoals, [ 'sell', 'promote' ] ).length > 0 ? 'business' : 'personal' );
+		const customerType = getSiteTypePropertyValue( 'slug', this.props.siteType, 'customerType' );
```
Comment (p-jackson) [inline client/signup/steps/plans/index.jsx]: Does this still need a default?

```suggestion
		const customerType = getSiteTypePropertyValue( 'slug', this.props.siteType, 'customerType' ) || 'personal';
```

File: client/signup/steps/plans/index.jsx [hunk 2]
```diff
@@ -4,7 +4,6 @@ import { Button } from '@automattic/components';
 import { isDesktop, subscribeIsDesktop, isMobile } from '@automattic/viewport';
 import classNames from 'classnames';
 import { localize } from 'i18n-calypso';
-import { intersection } from 'lodash';
```
Comment (p-jackson) [inline client/signup/steps/plans/index.jsx:7]: 😎 getting rid of some more `lodash` too 😎

**Review by p-jackson — COMMENTED** [review]
Looks good! Going to `/start/main` redirects to `/start` flow 👍

File: client/signup/steps/plans/index.jsx (hunk 1 above)
Comment (p-jackson) [reply client/signup/steps/plans/index.jsx]: I think it could be important because it might inform what plan is labelled the "recommended" one.

**Review by p-jackson — APPROVED** [review]

File: client/state/signup/steps/site-goals/actions.js [hunk 3]
```diff
@@ -1,10 +0,0 @@
-import { SIGNUP_STEPS_SITE_GOALS_SET } from 'calypso/state/action-types';
```
Comment (tyxla) [inline client/state/signup/steps/site-goals/actions.js:1]: We can also remove the `SIGNUP_STEPS_SITE_GOALS_SET` action type as it's no longer used.

**Review by tyxla — COMMENTED** [review]
It's wonderful to see some more dead code removal ❤️  Thank you!

**Pull request merged**